infrastructure/channels/serialize_child.html

```diff
@@ -0,0 +1,167 @@
+        throw new Error(`Constructors differ, expected ${expectedValue.constructor.name}, got ${actualValue.constructor.name}`);
+    }
+    if (expectedValue.constructor.name === "SendChannel") {
+        if (expectedValue.uuid !== actualValue.uuid) {
+            throw new Error(`SendChannels differ, expected uuid ${expectedValue.uuid}, got ${actualValue.uuid}`);
+        }
+    }
+    else if (expectedValue.constructor.name === "RegExp") {
+        if (expectedValue.source !== actualValue.source ||
+            expectedValue.flags !== actualValue.flags) {
+            throw new Error(`Regexps differ, expected ${expectedValue}, got ${actualValue}`);
+        }
+    } else if (expectedValue.constructor.name == "Date") {
+        if (expectedValue.toDateString() !== actualValue.toDateString()) {
+            throw new Error(`Dates differ, expected ${expectedValue.toDateString()}, got ${actualValue.toDateString()}`);
+        }
+    } else if (expectedValue instanceof Error) {
+        if (expectedValue.message !== actualValue.message ||
+            expectedValue.lineNumber !== actualValue.lineNumber ||
+            expectedValue.columnNumber !== actualValue.columnNumber ||
+            expectedValue.fileName !== actualValue.fileName) {
+            throw new Error(`Errors differ, expected ${expectedValue}, got ${actualValue}`);
+        }
+    } else if (Array.isArray(expectedValue)) {
+        seenActual = seen.has(actualValue);
+        seenExpected = seen.has(expectedValue)
+        if (seenActual && seenExpected) {
+            return;
+        } else if (seenExpected && !seenActual) {
+            throw new Error(`Expected cyclic array`);
+        } else if (!seenExpected && seenActual) {
+            throw new Error(`Got unexpected cyclic array`);
+        }
+        seen.add(actualValue);
+        seen.add(expectedValue);
+
+        if (actualValue.length !== expectedValue.length) {
+            throw new Error(`Array lengths differ, expected ${expectedValue.length}, got ${actualValue.length}`);
+        }
+        for (let i=0; i<actualValue.length; i++) {
+            compareValue(actualValue[i], expectedValue[i], seen);
+        }
+    } else if (expectedValue.constructor.name === "Set") {
+        seenActual = seen.has(actualValue);
+        seenExpected = seen.has(expectedValue)
+        if (seenActual && seenExpected) {
+            return;
+        } else if (seenExpected && !seenActual) {
+            throw new Error(`Expected cyclic set`);
+        } else if (!seenExpected && seenActual) {
+            throw new Error(`Got unexpected cyclic set`);
+        }
+        seen.add(actualValue);
+        seen.add(expectedValue);
+
+
+        if (actualValue.size !== expectedValue.size) {
+            throw new Error(`Set sizes differ, expected ${expectedValue.size}, got ${actualValue.size}`);
+        }
+        // So for a set we can't really check if the values are the same
```

Comment:
You mean because we can't sort them? I think we could check if they're the same, but it would be O(n^2), and with nested sets worse still.